### PR TITLE
Paperwork database + improvements + ACS

### LINKED ISSUE #1379: ACS pencode for SCP articles
[ACS explanation](https://scp-wiki.wikidot.com/anomaly-classification-system-guide)

Implement some way to generate ACS images, and add them to all SCP files.

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Adds the ability to use ordered lists with pencode.
Fixes a major bug with upload databases (when you downloaded a file, it would literally be the exact same file as on the database)
Fixes the formatting of SCP articles; they actually use pencode now, and thus won't break if you try and edit them.

Adds a second database to the server room: the Paperwork database. This database provides copies of document structures, allowing players to quickly use high-end forms without major effort. As the IT Technician can add new files to the database, this also provides some content for Office Workers.

P.S. if you have a form that you'd like to be added, simply ask me and I'll add it for you.

ALSO ALSO
ACS is finally here (thanks to checkraisefold for doing most of the work)
Closes #1379 

## Why It's Good For The Game

Bugfixes are good.

Having another server to take care of gives more content to the IT Technician, and having a centralized, easily accessible store of forms improves roleplay quality.

## Changelog

:cl: cheesePizza2, checkraisefold
add: Ordered lists were added to pencode. Check the codex for more info.
add: A new database was added to the server room - the Paperwork Database. Quickly download useful forms from the (relative) safety of your laptop! Office workers, consider sending the IT Tech new forms to add to the download list.
add: ACS is here! SCP articles now have a special ACS bar that displays info. Check the Pencode codex entry for details on usage.
fix: SCP articles actually use pencode now - you can edit them without breaking them.
fix: Downloaded files are now *copies* of the server's file, not actual references to their file. Whoops!
code: Paper uses /datum/browser, allowing for compatibility with newer Internet Explorer versions.
/:cl:

### CODE REVIEW

**Comment by checkraisefold** [comment]
change ulist back to list to avoid breaking existing pencode or have list as an alias to ulist

**Comment by aidan-bowman** [comment]
> change ulist back to list to avoid breaking existing pencode or have list as an alias to ulist

done

File: code/modules/paperwork/paper.dm [hunk 1]
```diff
@@ -188,9 +187,16 @@
...
 		html += "<hr/>" + info
-	html += "[stamps]</body></html>"
-	show_browser(user, html, "window=paper_[name]")
-	onclose(user, "paper_[name]")
+	html += "[stamps]"
+
+	// Ported to browser datum for IE11 feature parity
+	var/datum/browser/window = new(user, "paper_[name]")
+	window.stylesheets.Remove("common")
+	window.add_stylesheet("acs", 'html/acs.css')
```
Comment (checkraisefold) [inline code/modules/paperwork/paper.dm:195]: Should probably fix the hacky solution I did of networking the ACS stylesheet for every single piece of paper regardless of whether it uses ACS tags

File: code/modules/asset_cache/assets/tgui.dm [hunk 2]
```diff
@@ -3,6 +3,7 @@
 	assets = list(
 		"tgui.bundle.js" = file("tgui/public/tgui.bundle.js"),
 		"tgui.bundle.css" = file("tgui/public/tgui.bundle.css"),
+		"tgui.acs.css" = file("html/acs.css"),
```
Comment (checkraisefold) [inline code/modules/asset_cache/assets/tgui.dm:6]: This is also really stupid and horrible and really bad

File: code/_helpers/text.dm [hunk 3]
```diff
@@ -484,6 +489,104 @@
...
 
+/// Scans for and replaces pencode ACS formats with usable HTML versions
+/proc/pencode_acs2html(text)
+	// raw text for regex
+	var/regex/scanner_nosecondary = new(@"\[acs item_number=(\w+) clearance_level=(\w+) containment_class=(\w+) disruption_class=(\w+) risk_class=(\w+)\]", "g")
+	var/regex/scanner_secondary = new(@"\[acs item_number=(\w+) clearance_level=(\w+) containment_class=(\w+) secondary_class=(\w+) disruption_class=(\w+) risk_class=(\w+)\]", "g")
+
+	var/newtext = text
+
+	newtext = scanner_nosecondary.Replace(newtext,
```
Comment (checkraisefold) [inline code/_helpers/text.dm:500]: How slow are these regexes? Is this a new server crash method if I put a billion ACS tags in a paper?

File: code/modules/asset_cache/assets/tgui.dm (hunk 2 above)
Comment (aidan-bowman) [reply code/modules/asset_cache/assets/tgui.dm:6]: ghetto solution ftw

**Pull request merged**